server/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -778,21 +774,21 @@ public DiscoveryNodes nodes() {
             return nodes;
         }
 
-        public Builder putTransportVersion(String nodeId, TransportVersion version) {
-            transportVersions.put(nodeId, Objects.requireNonNull(version, nodeId));
+        public Builder putVersionsWrapper(String nodeId, VersionsWrapper versionsWrapper) {
```

Comment:
Pretty much every caller of this is doing `putVersionsWrapper(x, new VersionsWrapper(y))`. I suggest keeping this as `putTransportVersion(x, y)` and constructing the `VersionsWrapper` here instead. Then as we add versions we can add args to this method (and leave the method name alone until we add the next version).